packages/styled-components/src/utils/stylis.js

```diff
@@ -94,5 +88,11 @@ export default function createStylisInstance({
     _selectorRegexp = new RegExp(`\\${_selector}\\b`, 'g');
 
     return stylis(prefix || !selector ? '' : selector, cssStr);
-  };
+  }
+
+  // stringifying the function bodies is suboptimal, but some plugins
+  // are anonymous functions so there is no name to use as a token
+  stringifyRules.signature = plugins.join('');
```

Comment:
Very open to other suggestions... I'm not sure this is entirely safe since server vs client code can have different transpiled outputs and cause varying downstream hashes :/